composer/core/state.py

```diff
@@ -747,6 +814,27 @@ def load_optim_state(self, state_dict: Dict[str, Any]):
             else:
                 target.load_state_dict(source)
 
+    def _load_dataset_state(self, obj: Dict[str, Any]) -> None:
+        self.dataset_state = obj
+
+        dataset = self._dataset_of(self.train_dataloader)
+        if hasattr(dataset, 'load_state_dict'):
+            dataset.load_state_dict(obj['train'])  # pyright: ignore
+            obj['train'] = None
+            self.dataset_resumption['train'] = True
+
+        for evaluator in self.evaluators:
+            dataset = self._dataset_of(evaluator)
+            if hasattr(dataset, 'load_state_dict') and evaluator.label in obj['eval']:
+                dataset.load_state_dict(obj['eval'][evaluator.label])  # pyright: ignore
+                del obj['eval'][evaluator.label]
+                if 'eval' not in self.dataset_resumption:
+                    self.dataset_resumption['eval'] = {}
+                # Note: We currently disable setting dataset_resumption for eval datasets,
+                # which means they have one sample fetched in _spin_dataloaders before training
+                # starts. This avoids "CUDA error: initialization error" -- its not clear why.
+                # self.dataset_resumption['eval'][evaluator.label] = True
```

Comment:
Can we make sure we're tracking this somewhere in jira?

Reply:
done